### DISCUSSION TITLE
Schema types differ based on DB connection

### DISCUSSION THREAD

**Discussion by kburton-dev** [dedoc/scramble#278] [Q&A] [answered]
The schema types differ slightly when generating the OpenAPI json, when using different databases, such as MySQL and SQLite.

This is to be expected if the idea is to rely on the DB schema alone to generate the API schema, of models and json resources.

I am curious if it would be possible to use other analysis strategies as well, such as inspecting the model doc blocks, and "casts" property on the models.

![image](https://github.com/dedoc/scramble/assets/10101978/31125951-7fce-4de3-9980-10d94502ae67)

**Comment by romalytvynenko** [comment]
Hey @kburton-dev 

Can you check this issue on version `0.9.0`? Should be fixed now

**Reply by kburton-dev** [reply]
It did not work for me.

However, I can fix it by updating the `ModelExtension::getBaseAttributeType()` method to handle casts. Just before the last return statement, resolving casted values like this can work:

```
$eloquentCastedType = match ($value['cast']) {
    'array', 'json' => new ArrayType(),
    'real', 'float', 'double' => new FloatType(),
    'int', 'integer' => new IntegerType(),
    'bool', 'boolean' => new BooleanType(),
    ......
    default => null,
};

return $eloquentCastedType ?? $attributeType;
```

I can make a PR with this if you agree it's a reasonable solution. I will add the remaining cast types and update the tests.

**Reply by kburton-dev** [reply, marked as answer]
PR over here https://github.com/dedoc/scramble/pull/345
